pkg/inject/sidecar.go

```diff
@@ -80,10 +80,10 @@ func Desired(dep *appsv1.Deployment, ns *corev1.Namespace) bool {
 		"namespace":  dep.Namespace,
 		"deployment": dep.Name,
 	})
-	_, depExist := dep.Annotations[Annotation]
+	annotationValue, depExist := dep.Annotations[Annotation]
 	_, nsExist := ns.Annotations[Annotation]
 
-	if depExist {
+	if depExist && !strings.EqualFold(annotationValue, "false") {
```

Comment:
What if the annotation is at the namespace level? Isn't this logic already present in the `inject.Desired`?

Reply:
inject.Desired just test if the annotation is present, but if the annotation exists with a "false" value it still returns "true".

Should i validate the same condition for the namespace?

Reply:
I'm confident we have a logic somewhere that determines whether a deployment should have a sidecar based on annotations from both the deployment itself and the namespace.

Reply:
it's present in "inject.select", but doesn't it make more sense for this to be validated in "inject.desired" ?, it is a bit confusing that desired returns "true" when it is not really desired.

Reply:
True. Would you be able to create an issue to refactor that? Also, it would be great to have an e2e test to assert the behavior of this PR here. It doesn't have to be done as part of this PR, though.

I'd still like to get @rubenvp8510's review before merging, but looks good to me!

Reply:
> True. Would you be able to create an issue to refactor that? Also, it would be great to have an e2e test to assert the behavior of this PR here. It doesn't have to be done as part of this PR, though.

count on it

Reply:
This covers both cases,  because `inject.Needed ` is used when we modify the namespace to determine if the deployments that belongs to namespace needs an injection. Same case for individual deployments.

Except that we already have the cleaning logic on namespace  but not on individual deployments (which is something this PR adds )

I think this modification to the desired behaviour makes sense , but it is a little bit confusing the difference between needed and desired,  Indeed `inject.Needed` calls` inject.Desired.` so I think is enough to expose `Needed` and use it on the controllers.

Reply:
BTW, I think we shouldn't use select for validating these cases, inject.Select  is just when we are make sure we need to inject something but we need to find which instance we need to inject (and if it exists)

Reply:
the refactor is done, can you review @rubenvp8510 ?